pkg/lspext/partial.go

```diff
@@ -0,0 +1,15 @@
+package lspext
+
+import "github.com/sourcegraph/go-langserver/pkg/lsp"
+
+// PartialResultParams is the input for "$/partialResult", a notification.
+type PartialResultParams struct {
+	ID      lsp.ID      `json:"id"`
+	Patches []JSONPatch `json:"patches"`
+}
+
+type JSONPatch struct {
```

Comment:
We might wanna move these out into a different file for JSON Patch.
have you looked at https://github.com/evanphx/json-patch and https://github.com/mattbaird/jsonpatch?